src/main/java/xyz/aadev/generitech/common/blocks/power/BlockPowerStorage.java

```diff
@@ -0,0 +1,142 @@
+
+    public BlockPowerStorage() {
+        super(Material.ROCK, "machines/pulverizer/pulverizer", MachineTier.all());
+        this.setDefaultState(blockState.getBaseState().withProperty(MACHINETIER, MachineTier.TIER_0));
+        this.setTileEntity(TileEntityPowerStorage.class);
+        this.setCreativeTab(GeneriTechTabs.GENERAL);
+        this.setInternalName("powerstorage");
+    }
+
+
+    @Override
+    public boolean onBlockActivated(World world, BlockPos pos, IBlockState state, EntityPlayer player, EnumHand hand, ItemStack heldItem, EnumFacing side, float hitX, float hitY, float hitZ) {
+        TileEntity tileEntity;
+        tileEntity = world.getTileEntity(pos);
+        GeneriTech.Logger.info(Long.toString(((TileEntityPowerStorage) tileEntity).getStoredPower()));
+
+        if (!world.isRemote) {
+
+            player.openGui(GeneriTech.getInstance(), Reference.GUI_ID.POWERSTORAGE_GUI, world, pos.getX(), pos.getY(), pos.getZ());
+
+
+        }
+        return true;
+    }
+
+    @Override
+    public IBlockState getActualState(IBlockState state, IBlockAccess worldIn, BlockPos pos) {
+        TileEntityPowerStorage tileEntity = TileHelper.getTileEntity(worldIn, pos, TileEntityPowerStorage.class);
+        if (tileEntity != null && tileEntity.canBeRotated()) {
+            return state.withProperty(FACING, tileEntity.getForward());
+        }
+        return state.withProperty(FACING, EnumFacing.NORTH);
+    }
+
+    @Override
+    public int getLightValue(IBlockState state, IBlockAccess world, BlockPos pos) {
+        IBlockState blockState = getActualState(state, world, pos);
+        return (blockState.getValue(MACHINETIER) == MachineTier.TIER_0) ? 15 : 0;
+    }
+
+    @Override
+    protected BlockStateContainer createBlockState() {
+        return new BlockStateContainer(this, MACHINETIER, FACING);
+    }
+
+    @Override
+    public int damageDropped(IBlockState state) {
+        return getMetaFromState(state);
+    }
+
+    @Override
+    public void breakBlock(World world, BlockPos blockPos, IBlockState blockState) {
+        TileEntityPowerStorage tileEntity = TileHelper.getTileEntity(world, blockPos, TileEntityPowerStorage.class);
+
+        TileHelper.DropItems(tileEntity, 0, 0);
+
+    }
+
+    @Override
+    public void randomDisplayTick(IBlockState state, World worldIn, BlockPos pos, Random rand) {
+        TileEntityPowerStorage tileEntity = TileHelper.getTileEntity(worldIn, pos, TileEntityPowerStorage.class);
+        if (tileEntity == null)
+            return;
+
+        EnumFacing enumfacing = tileEntity.getForward();
+        double d0 = (double) pos.getX() + 0.5D;
+        double d1 = (double) pos.getY() + rand.nextDouble() * 6.0D / 16.0D;
+        double d2 = (double) pos.getZ() + 0.5D;
+        double d3 = 0.52D;
+        double d4 = rand.nextDouble() * 0.6D - 0.3D;
+
+        EnumParticleTypes particleTypes = null;
+        switch (tileEntity.getBlockMetadata()) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Replace this "switch" statement by "if" statements to increase readability. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AS1301)